x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -362,19 +364,29 @@ export const TimelineDataTableComponent: React.FC<DataTableProps> = memo(
       return enabledRowRenderers.length > 0 ? trailingControlColumns : undefined;
     }, [enabledRowRenderers.length, trailingControlColumns]);
 
+    const gridStyles = useMemo(() => {
+      const rowClasses = tableRows.reduce((acc: Record<number, string>, item, index) => {
+        acc[index] = getEventTypeRowClassName(item.ecs);
+        return acc;
+      }, {});
+
+      return { ...GRID_STYLE, rowClasses };
+    }, [tableRows]);
```

Comment:
Although this works I am trying to think if we should combine the `transformTimelineItemToUnifiedRows` and this so that we do not have to loop twice but I think it is tough to avoid. I will give it some more thought today.

Reply:
Done!